roles/openshift_master/tasks/systemd_units.yml

```diff
@@ -38,32 +38,81 @@
   when: create_ha_unit_files | changed
 # end workaround for missing systemd unit files
 
+- name: Preserve Master API Proxy Config options
+  command: grep PROXY /etc/sysconfig/{{ openshift.common.service_type }}-master-api
+  register: master_api_proxy
+  when: openshift.master.ha is defined and openshift.master.ha | bool and openshift_master_cluster_method == "native"
+      and upgrading is defined and upgrading | bool
```

Comment:
@dgoodwin are you ok with my use of upgrading variable here? That's not going away in your in-progress work is it?